### PR TITLE
#9 Add community material design icons

### LINKED ISSUE #9: Add community material design icons for use in the Icon components
In web foundry, we can pass in mdi______ variables which contain the path attribute of an svg - which get passed to the "Icon" component from @mdi/react which just stuffs the path into an svg. We used that in button and text to add iconography support.

I've added the community material design icons to an RN project before but I don't remember how it gets implemented in the code. I'm thinking the API may have to change for native to accommodate the new way of getting icons (icon web fonts)

AC:
* Add the community material design icons to foundry-native-ui
* Enable them in the API for Text and Button with as little change to the API as possible

### PR DESCRIPTION
**Pull request by dawsonbooth**
- Adds an icon component that is similar to that of [`@mdi/react`](https://github.com/Templarian/MaterialDesign-React)
- Takes in and renders any SVG path through the `path` prop
- Uses `Animated` to optionally spin the icons with a duration (useful for `isProcessing`)

### CODE REVIEW

File: src/components/Icon/Icon.tsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    ).start();
+  }, [spinAnim]);
+
+  const interpolatedSpinAnim = spinAnim.interpolate({
+    inputRange: [0, 100],
+    outputRange: ['0deg', '360deg'],
+  });
+
+  return (
+    <Animated.View style={[{ transform: [{ rotate: interpolatedSpinAnim }] }]}>
```
Comment (aVileBroker) [inline src/components/Icon/Icon.tsx:50]: We should use the AnimatedView from the base elements (which are styled by the globalstyles)

**Review by aVileBroker — CHANGES_REQUESTED** [review]
Looks great! Had one comment but I'm glad you got the original API working!

File: src/components/Icon/Icon.tsx (hunk 1 above)
Comment (dawsonbooth) [reply src/components/Icon/Icon.tsx]: Good catch!

File: package.json [hunk 2]
```diff
@@ -120,7 +121,9 @@
     "react-native-reanimated": "~2.1.0",
     "react-native-safe-area-context": "3.2.0",
     "react-native-screens": "~3.0.0",
-    "react-native-web": "~0.13.12"
+    "react-native-svg": "^12.1.1",
+    "react-native-web": "~0.13.12",
+    "react-spring": "^9.2.3"
```
Comment (aVileBroker) [inline package.json:126]: Seems like we're not using react-spring

I know we'll use it eventually if you stick with the react-native Animated class but just wanted to point it out either way.

**Review by aVileBroker — APPROVED** [review]

**Pull request merged**